object_store/src/gcp/mod.rs

```diff
@@ -790,12 +803,24 @@ impl GoogleCloudStorageBuilder {
             service_account_path,
             client,
             retry_config,
+            proxy_url,
         } = self;
 
         let bucket_name = bucket_name.ok_or(Error::MissingBucketName {})?;
         let service_account_path =
             service_account_path.ok_or(Error::MissingServiceAccountPath)?;
-        let client = client.unwrap_or_else(Client::new);
+        let client = client.unwrap_or(match proxy_url {
```

Comment:
```suggestion
        let client = client.unwrap_or_else(|| match proxy_url {
```

It's a minor nit, but this avoids creating a client eagerly, only to just discard it